components/autocomplete/index.js

```diff
@@ -225,22 +225,44 @@ export class Autocomplete extends Component {
 		this.node = node;
 	}
 
-	insertCompletion( range, replacement ) {
-		const container = document.createElement( 'div' );
-		container.innerHTML = renderToString( replacement );
-		while ( container.firstChild ) {
-			const child = container.firstChild;
-			container.removeChild( child );
-			range.insertNode( child );
-			range.setStartAfter( child );
-		}
+	insertCompletion( range, replacement, completerName ) {
+		// Wrap completions so we can treat them as tokens with editing boundaries.
+		const tokenWrapper = document.createElement( 'span' );
+		tokenWrapper.classList.add( 'autocomplete-token' );
+		tokenWrapper.classList.add( `autocomplete-token-${ completerName }` );
+
+		tokenWrapper.innerHTML = renderToString( replacement );
+
+		range.insertNode( tokenWrapper );
```

Comment:
Wondering if we should move this to dom utils and keep all range logic there...

Reply:
I feel hesitant to move this to the dom utils because I dislike the `&nbsp;` compromise so much :-P

But if we were to move this to dom utils, what do you imagine we'd create there? Something like this?

```typescript
// Please feel free to suggest improved name.
export function replaceRangeWithToken( range : Range, value : <string|WPElement>, tokenClasses : Array<string> ) {
  const tokenWrapper = document.createElement( 'span' );
  tokenClasses.forEach( ( c ) => tokenWrapper.classList.add( c ) );
  tokenWrapper.innerHTML = renderToString( replacement );
  
  range.insertNode( tokenWrapper );
  range.setStartAfter( tokenWrapper );
  range.deleteContents();
  
  // <comment explaining why nbsp>
  tokenWrapper.parentNode.insertBefore(
  	document.createTextNode( '\u00A0' ),
  	tokenWrapper.nextSibling
  );
  
  const selection = window.getSelection();
  selection.removeAllRanges();
  
  const newCursorPosition = document.createRange();
  newCursorPosition.setStartAfter( tokenWrapper.nextSibling );
  selection.addRange( newCursorPosition );
}
```

I feel like the above may be too specific to be valuable for reuse, but it _would_ be nice to keep Range-related logic in the dom utils.